feat(workflow): Including GroupOwner in Group serializer

iProgramStuff opened the pull request:
Adding support for a new `expand` attribute in the `StreamGroupSerializerSnuba` serializer: `owners`.

Using `expand=owners` will now attach GroupOwner information as a list of dicts. Each dict will contain `type`, `owner`, and `date_added`.

`type` will be a string from `GROUP_OWNER_TYPE`
`owner` will be a string of format `owner_type:id`, e.g. `team:12`, `user:41`

iProgramStuff committed:
Including GroupOwner in Group serializer

davidenwang commented on src/sentry/api/endpoints/organization_group_index.py:
```diff
@@ -115,6 +115,9 @@ def get(self, request, organization):
         collapse = request.GET.getlist("collapse", [])
         has_inbox = features.has("organizations:inbox", organization, actor=request.user)
+        has_workflow_owners = features.has(
+            "organizations:workflow-owners", organization, actor=request.user
```
do we need to add this to the features `default_manager`?

iProgramStuff replied:
Yeah :)
https://github.com/getsentry/sentry/pull/22241/

davidenwang replied:
oh my b didn't see that

iProgramStuff replied:
I added it after your comment 😎

davidenwang replied:
oh haha i take it back then 😎

iProgramStuff committed:
Merge remote-tracking branch 'origin' into feat/owneringroupserializer

markstory commented on src/sentry/models/groupowner.py:
```diff
@@ -57,3 +57,18 @@ def owner(self):
+    group_owners = GroupOwner.objects.filter(group__in=group_ids)
+    owner_details = {
+        go.group_id: {
+            "type": go.type,
```
Should we convert the type back to the string name? We often convert integer enums back to string values like in transaction.status, activity types, group subscription reasons and more. The string values are simpler for end users as they don't need to refer to an enum to know what the current value is.

iProgramStuff replied:
Will do, thanks!

markstory commented on tests/snuba/api/endpoints/test_organization_group_index.py:
```diff
@@ -739,21 +741,8 @@ def test_inbox_fields(self):
-            query += u" active_at:" + iso_format(before_now(seconds=350))
-            query += u" first_seen:" + iso_format(before_now(seconds=500))
-
+            query = u" status:unresolved"
```
```suggestion
            query = u"status:unresolved"
```

dashed committed:
ref(dashboard): Modify "Revert" button to become a "Cancel" button (#22265)

iProgramStuff committed:
Update tests/snuba/api/endpoints/test_organization_group_index.py

Co-authored-by: Mark Story <mark@sentry.io>

robinrendle committed:
ref(images loaded): Removed Clippedbox (#22181)

tonyo committed:
fix: Temporary disable reprocessing2 (#22278)

markstory committed:
feat(dashboards) Improve validation for dashboard update (#22239)

Add validation for field list, conditions and intervals on queries.

BYK committed:
ci(release): Fetch the last 20 commits, not just 1 (#22152)

This should fix the merge error we encountered in the last release: https://github.com/getsentry/sentry/runs/1413845407?check_suite_focus=true#step:10:41

doralchan committed:
feat(update tag): web vitals tag (#22269)

* update tags

* red tag

Co-authored-by: Dora Chan <dora.lchan@gmail.com>

Zylphrex committed:
fix(spans): Format size data in bytes (#22261)

Size information was formatted as a number in the span details. This is not
easy to read, instead, format them in bytes for better legibility.

scttcper committed:
fix(workflow): Check query type in discover query (#22282)

scttcper committed:
fix(workflow): Tweak inbox stream styles, add inbox storybook (#22279)

dashed committed:
ref(dashboard): Add line and bar charts to Dashboard widget (#22271)

iProgramStuff committed:
Changing type to string

iProgramStuff committed:
Merge remote-tracking branch 'origin' into feat/owneringroupserializer

iProgramStuff committed:
Converting enum value to string

iProgramStuff committed:
updating test

davidenwang commented on tests/snuba/api/endpoints/test_organization_group_index.py:
```diff
@@ -739,21 +742,8 @@ def test_inbox_fields(self):
                 data={"timestamp": iso_format(before_now(seconds=500)), "fingerprint": ["group-1"]},
                 project_id=self.project.id,
             )
-            self.store_event(
```
just curious cuz i don't touch the backend as often, but were these not needed?

iProgramStuff replied:
Yeah they were totally irrelevant to what was actually being tested and seemed fine to be removed (...aka I don't know why I put them there in the first place 😄 )

davidenwang reviewed (approved):
lgtm

iProgramStuff commented on src/sentry/models/groupowner.py:
```diff
@@ -15,6 +15,12 @@ class GroupOwnerType(Enum):
 
 
+GROUP_OWNER_TYPE = {
+    GroupOwnerType.SUSPECT_COMMIT: "Suspect Commit",
+    GroupOwnerType.OWNERSHIP_RULE: "Ownership Rule",
```
So I'm sending "Suspect Commit"/"Ownership Rule". Not sure if the frontend would rather something like "suspect_commit" / "ownership_rule"?

davidenwang replied:
should be fine on the frontend either way, not sure if there is a standard tho for snakecase :/

markstory replied:
Many of the enums we have in the UI today use `camelBack` form. So `suspectCommit` and `ownershipRule` would work.

iProgramStuff replied:
Changing to that 👍

iProgramStuff committed:
Changing to camelCase

wedamija commented:
@iProgramStuff Since this returns a list of objects, does it make more sense for this to be a separate endpoint? `groups/1/owners` or something along those lines?
